### ISSUE TITLE
error with specific form of array construction for all types

### ISSUE THREAD

**Issue by tserg** [vyperlang/vyper#2587]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.1
* OS: linux
* Python Version (output of `python --version`): 3.8.5

### What's your issue about?

The following code snippet compiles:
```
a: bool
b: bool

@external
def __init__():
    self.a = True
    self.b = False

@external
def doThings():
    assert True in [self.a, self.b]
```
but this gives an error: `KeyError: 'type'`
```
a: bool
b: bool

@external
def __init__():
    self.a = True
    self.b = False

@external
def doThings():
    for c in [self.a, self.b]:
        pass
```
If we change it to this, then it compiles too:
```
@external
def doThings():
    a: bool = True
    b: bool = False
    for c in [a, b]:
        pass
```

The error is caused by this line:
https://github.com/vyperlang/vyper/blob/2f1584bf9d2fce15e2173b8befc83619c5deae79/vyper/old_codegen/expr.py#L408

### How can it be fixed?

Not sure, still looking into it.

**Comment by charles-cooper** [comment]
I think this might be fixed in #2556

**Comment by tserg** [comment]
> I think this might be fixed in #2556

Yes it does!
